packages/webapp/src/components/Forms/Sale/index.jsx

```diff
@@ -1,85 +1,244 @@
-import React from 'react';
-import footerStyles from '../../LogFooter/styles.module.scss';
+import React, { useState, useEffect, useRef } from 'react';
 import { useTranslation } from 'react-i18next';
+import Form from '../../Form';
+import Button from '../../Form/Button';
+import Input, { getInputErrors } from '../../Form/Input';
+import Unit from '../../Form/Unit';
+import { useForm } from 'react-hook-form';
+import PageTitle from '../../PageTitle/v2';
+import FilterPillSelect from '../../Filter/FilterPillSelect';
+import { Error } from '../../Typography';
+import { harvestAmounts } from '../../../util/convert-units/unit';
+import { getDateInputFormat } from '../../../util/moment';
 
 const SaleForm = ({
-  currencySymbol,
   cropVarietyOptions,
-  chosenOptions,
-  handleChooseCrop,
-  model,
   onSubmit,
-  quantityUnit,
-  footerText,
-  footerOnClick,
+  onClickDelete,
+  title,
+  dateLabel,
+  customerLabel,
+  system,
+  currency,
+  sale,
 }) => {
+  // Reformat selector to match component format
+  // TODO: match component to selector format
+  const existingSales = sale?.crop_variety_sale.reduce(
+    (acc, cur) => ({
+      ...acc,
+      [cur.crop_variety_id]: {
+        crop_variety_id: cur.crop_variety_id,
+        quantity: cur.quantity,
+        quantity_unit: cur.quantity_unit,
+        sale_value: cur.sale_value,
+      },
+    }),
+    {},
+  );
+  // FilterPillSelect details
+  const STATUS = 'STATUS';
+  const filterRef = useRef({});
+  const filter = {
+    filterKey: STATUS,
+    options: cropVarietyOptions.map((cvs) => ({
+      value: cvs.label,
+      default: existingSales?.[cvs.value] ? true : false,
+      label: cvs.label,
+    })),
+  };
+  // Unique hook form names
   const { t } = useTranslation();
+  const SALE_DATE = 'sale_date';
+  const SALE_CUSTOMER = 'customer_name';
+  const CHOSEN_VARIETIES = 'crop_variety_sale';
+  let CROP_VARIETY_VALUES = {};
+  cropVarietyOptions.forEach((cv) => {
+    CROP_VARIETY_VALUES[cv.value] = {
+      CROP_VARIETY_ID: `${CHOSEN_VARIETIES}.${cv.value}.crop_variety_id`,
+      SALE_VALUE: `${CHOSEN_VARIETIES}.${cv.value}.sale_value`,
+    };
+  });
+
+  const {
+    register,
+    handleSubmit,
+    watch,
+    getValues,
+    setValue,
+    control,
+    formState: { errors, isValid },
+  } = useForm({
+    mode: 'onChange',
+    shouldUnregister: true,
+    defaultValues: {
+      [SALE_DATE]: getDateInputFormat(sale?.sale_date) || getDateInputFormat(),
+      customer_name: sale?.customer_name,
+      crop_variety_sale: existingSales ?? undefined,
+    },
+  });
+
+  const saleDateRegister = register(SALE_DATE, { required: true });
+  const saleCustomerRegister = register(SALE_CUSTOMER, { required: true });
+  // FilterPillSelect does not support register yet, so const definition not needed
+  register(CHOSEN_VARIETIES, { required: true });
+
+  const [isDirty, setIsDirty] = useState(false);
+  const [filterState, setFilterState] = useState(filter);
```

Comment:
Is this initial filterState (filter) used anywhere? Can't this be useState({})?

Reply:
Thank you, there was a time I thought this was necessary to get the filter functioning properly. I still don't 100% understand the pill filter. But yes it seems I can now remove that thanks!